Type errors with Fab and Button components from core when installing latest versions of material-ui/core

Seeing type errors with the Button and Fab components when installing the latest version of material-ui/core

- [x] I have searched the [issues](https://github.com/mui-org/material-ui/issues) of this repository and believe that this is not a duplicate.

## Expected Behavior 🤔

No type errors

## Current Behavior 😯

```
node_modules/@material-ui/core/Button/Button.d.ts:5:18 - error TS2430: Interface 'ButtonProps' incorrectly extends interface 'Pick<ButtonBaseProps, "disabled" | "media" | "hidden" | "dir" | "form" | "slot" | "style" | "title" | "value" | "id" | "onError" | "name" | "type" | "color" | "target" | "download" | ... 262 more ... | "TouchRippleProps"> & StyledComponentProps<...> & { ...; }'.
  Type 'ButtonProps' is not assignable to type 'Pick<ButtonBaseProps, "disabled" | "media" | "hidden" | "dir" | "form" | "slot" | "style" | "title" | "value" | "id" | "onError" | "name" | "type" | "color" | "target" | "download" | ... 262 more ... | "TouchRippleProps">'.
    Types of property 'type' are incompatible.
      Type 'string | undefined' is not assignable to type '"button" | "reset" | "submit" | undefined'.
        Type 'string' is not assignable to type '"button" | "reset" | "submit" | undefined'.

5 export interface ButtonProps extends StandardProps<ButtonBaseProps, ButtonClassKey, 'component'> {
                   ~~~~~~~~~~~

node_modules/@material-ui/core/Fab/Fab.d.ts:5:18 - error TS2430: Interface 'FabProps' incorrectly extends interface 'Pick<ButtonBaseProps, "disabled" | "media" | "hidden" | "dir" | "form" | "slot" | "style" | "title" | "value" | "id" | "onError" | "name" | "type" | "color" | "target" | "download" | ... 262 more ... | "TouchRippleProps"> & StyledComponentProps<...> & { ...; }'.
  Type 'FabProps' is not assignable to type 'Pick<ButtonBaseProps, "disabled" | "media" | "hidden" | "dir" | "form" | "slot" | "style" | "title" | "value" | "id" | "onError" | "name" | "type" | "color" | "target" | "download" | ... 262 more ... | "TouchRippleProps">'.
    Types of property 'type' are incompatible.
      Type 'string | undefined' is not assignable to type '"button" | "reset" | "submit" | undefined'.
        Type 'string' is not assignable to type '"button" | "reset" | "submit" | undefined'.

5 export interface FabProps extends StandardProps<ButtonBaseProps, FabClassKey, 'component'> {
                   ~~~~~~~~
```

## Steps to Reproduce 🕹

Apologies I don't have a live example, but I just installed the latest version in my app and the issue appears

## Context 🔦



## Your Environment 🌎

| Tech         | Version |
|--------------|---------|
| Material-UI  | v3.9.2  |
| React        |     16.8.5   |
| TypeScript   |   3.3.4000      |

--- a/packages/material-ui/src/Button/Button.d.ts
+++ b/packages/material-ui/src/Button/Button.d.ts
@@ -11,7 +11,7 @@ declare const Button: ExtendButtonBase<{
     fullWidth?: boolean;
     href?: string;
     size?: 'small' | 'medium' | 'large';
-    type?: string;
+    type?: 'submit' | 'reset' | 'button';
     variant?: 'text' | 'outlined' | 'contained';
   };
   defaultComponent: 'button';

--- a/packages/material-ui/src/Fab/Fab.d.ts
+++ b/packages/material-ui/src/Fab/Fab.d.ts
@@ -10,7 +10,7 @@ declare const Fab: ExtendButtonBase<{
     disableRipple?: boolean;
     href?: string;
     size?: 'small' | 'medium' | 'large';
-    type?: string;
+    type?: 'submit' | 'reset' | 'button';
     variant?: 'round' | 'extended';
   };
   defaultComponent: 'button';

